weed/s3api/chunked_reader_v4_test.go

```diff
@@ -53,13 +61,110 @@ func NewRequest() (*http.Request, error) {
 	return req, nil
 }
 
-func TestNewSignV4ChunkedReader(t *testing.T) {
-	req, err := NewRequest()
+func TestNewSignV4ChunkedReaderstreamingAws4HmacSha256Payload(t *testing.T) {
+	// This test will implement the following scenario:
+	// https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-streaming.html#example-signature-calculations-streaming
+	req, err := NewRequeststreamingAws4HmacSha256Payload()
 	if err != nil {
 		t.Fatalf("Failed to create request: %v", err)
 	}
+	iam := setupIam()
+
+	// The expected payload a long string of 'a's
+	expectedPayload := strings.Repeat("a", 66560)
+
+	runWithRequest(iam, req, t, expectedPayload)
+}
+
+func generateStreamingUnsignedPayloadTrailerPayload(includeFinalCRLF bool) string {
+	// This test will implement the following scenario:
+	// https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html
+
+	chunk1 := "2000\r\n" + strings.Repeat("a", 8192) + "\r\n"
+	chunk2 := "2000\r\n" + strings.Repeat("a", 8192) + "\r\n"
+	chunk3 := "400\r\n" + strings.Repeat("a", 1024) + "\r\n"
+
+	chunk4 := "0\r\n" /* the last chunk is empty */
+
+	if includeFinalCRLF {
+		// Some clients omit the final CRLF, so we need to test that case as well
+		chunk4 += "\r\n"
+	}
+
+	data := strings.Repeat("a", 17408)
+	writer := crc32.NewIEEE()
+	_, err := writer.Write([]byte(data))
+
+	if err != nil {
+		fmt.Println("Error:", err)
+	}
+	checksum := writer.Sum(nil)
+	base64EncodedChecksum := base64.StdEncoding.EncodeToString(checksum)
+	trailer := "x-amz-checksum-crc32:" + base64EncodedChecksum + "\n\r\n\r\n\r\n"
+	fmt.Println("Trailer:", trailer)
 
+	payload := chunk1 + chunk2 + chunk3 + chunk4 + trailer
+	return payload
+}
+
+func NewRequestStreamingUnsignedPayloadTrailer(includeFinalCRLF bool) (*http.Request, error) {
```

Comment:
FYI, this test does not include an `Authorization` header, so we do not check the signature, but clients do send it, and I've tested manually that it is indeed working.